### PR TITLE
Jsc/csv import

### LINKED ISSUE #355: Bulk load of keys
From csv file

### PR DESCRIPTION
**Pull request by jcubbage**
Closes #355 
First pass. Needs work for sure. Probably pull most of the code out of controller into service. But, I wanted a initial thoughts.

### CODE REVIEW

File: Keas.Mvc/Controllers/TeamAdminController.cs [hunk 1]
```diff
@@ -363,5 +366,102 @@ public async Task<IActionResult> RemovePpsDepartment(int id, TeamPpsDepartment p
...
+                Message = "File not selected";
+                return RedirectToAction("Upload");
+            }
+            // Add counts
+            var team = await _context.Teams.FirstAsync(t => t.Slug == Team);
+            using (var reader = new StreamReader(file.OpenReadStream()))
+            using (var csv = new CsvReader(reader))
+            {    
+               csv.Configuration.PrepareHeaderForMatch = (string header, int index) => header.ToLower();
+                 var record = new KeyImport();
```
Comment (jSylvestre) [inline Keas.Mvc/Controllers/TeamAdminController.cs]: This indent...

File: Keas.Mvc/Controllers/TeamAdminController.cs [hunk 2]
```diff
@@ -363,5 +369,172 @@ public async Task<IActionResult> RemovePpsDepartment(int id, TeamPpsDepartment p
...
+                                serial = new KeySerial();
+                                serial.Number = r.SerialNumber;
+                                serial.Name = r.SerialNumber;
+                                serial.Key = key;
+                                serial.Status = "Active";
+                                if(r.Status == "L" || r.Status == "Lost")
+                                {
+                                    serial.Status = "Lost";
+                                }
+                                if(r.Status == "B" || r.Status == "Destroyed")
```
Comment (jSylvestre) [inline Keas.Mvc/Controllers/TeamAdminController.cs]: @jcubbage We shouldn't be translating "their" codes. They should replace "B" in their file.

File: Keas.Mvc/Models/KeyImport.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    [Required]
+    public string Keynumber { get; set; }
+    [Required]
+    public string  SerialNumber { get; set; }
+    public string Description { get; set; }
+
+    public string KerbUser { get; set; }
+
+    public DateTime DateIssued { get; set; }
```
Comment (jSylvestre) [inline Keas.Mvc/Models/KeyImport.cs]: @jcubbage Needs to be nullable and you need to validate when you read it. (HasValue, isn't in the future, dueDate isn't in the past, etc.)

File: Keas.Mvc/Services/IdentityService.cs [hunk 4]
```diff
@@ -22,19 +22,82 @@ public interface IIdentityService
...
         private readonly ApplicationDbContext _context;
+    
 
         public IdentityService(IOptions<AuthSettings> authSettings, ApplicationDbContext context)
         {
             _authSettings = authSettings.Value;
             _context = context;
         }
 
+        public async Task<(Person Person, int peopleCount)> GetOrCreatePersonFromKerberos(string kerb, int teamId)
```
Comment (srkirkland) [inline Keas.Mvc/Services/IdentityService.cs]: Tuples?  Tuples!

**Comment by jSylvestre** [comment]
@jcubbage KeySerialAssignments. You are populating ApprovedAt (I don't think we are using this, but it is fine)

Keys.Tags is NULL instead of an empty string. (Might be a problem)

**Comment by jSylvestre** [comment]
@srkirkland This failed on appveyor because "don't" in the commit message

**Pull request merged**